diff --git a/src/game/memory.c b/src/game/memory.c
index 47295dc..e8fb088 100644
--- a/src/game/memory.c
+++ b/src/game/memory.c
@@ -361,7 +361,7 @@ void *load_segment_decompress(s32 segment, u8 *srcStart, u8 *srcEnd) {
             decompress(compressed, dest);
 #endif
 			osSyncPrintf("end decompress\n");
-            set_segment_base_addr(segment, dest);
+            set_segment_base_addr(segment, dest); sSegmentROMTable[segment] = (uintptr_t) srcStart;
             main_pool_free(compressed);
         } else {
         }
@@ -396,7 +396,7 @@ void *load_segment_decompress_heap(u32 segment, u8 *srcStart, u8 *srcEnd) {
 #elif MIO0
         decompress(compressed, gDecompressionHeap);
 #endif
-        set_segment_base_addr(segment, gDecompressionHeap);
+        set_segment_base_addr(segment, gDecompressionHeap); sSegmentROMTable[segment] = (uintptr_t) srcStart;
         main_pool_free(compressed);
     } else {
     }
